core/src/main/scala/org/apache/spark/memory/MemoryManager.scala

```diff
@@ -180,6 +180,26 @@ private[spark] abstract class MemoryManager(
     onHeapStorageMemoryPool.memoryUsed + offHeapStorageMemoryPool.memoryUsed
   }
 
+  /**
+   *  On heap execution memory currently in use, in bytes.
+   */
+  final def onHeapExecutionMemoryUsed: Long = onHeapExecutionMemoryPool.memoryUsed
```

Comment:
Since it's access just one Long field, I don't think so. The other methods, which are synchronized are summing 2 fields.

Reply:
It probably should be if only because the variable is annotated with `@GuardedBy(this)`, so it makes the code more consistent to mark this as synchronized.

Reply:
That's true -- I'll change the methods to synchronized.